fraqbot/Local/randomemoji.py

```diff
@@ -47,42 +47,71 @@ def _fetch_slack_emojis(self):
                 'emoji'
             )
 
-    def _get_emoji(self, how_many):
+    def _get_emoji(self, how_many, search_term):
         how_many_limited = max(
             self.min_how_many,
             min(self.max_how_many, how_many)
         )
+        search_term_normalized = (search_term
+                                  if search_term
+                                  and len(search_term) > 0
+                                  else None)
 
         emoji_response = self._fetch_slack_emojis()
 
+        emoji_list = list(emoji_response.keys())
+
+        if (search_term_normalized):
+            filtered_emoji_list = [
+                emoji_name for emoji_name
+                in emoji_list
+                if search_term_normalized in emoji_name
+            ]
+            if len(filtered_emoji_list) < 1:
+                return ('Nothing matched search term. '
+                        + 'Please accept this instead: :'
+                        + random.choice(emoji_list) + ':')
+            else:
+                emoji_list = filtered_emoji_list
+
         return (':'
                 + ': :'.join(random.choices(
-                        list(emoji_response.keys()), k=how_many_limited
+                        emoji_list, k=how_many_limited
                     )).strip()
                 + ':')
 
     def handle(self, message):
         logger.debug(
             'Handling Random Emoji request: {}'.format(message['text'])
         )
-        text_provided = message['text'][6:].strip()
-
         opts = self.build_reply_opts(message)
-        if len(text_provided) > 0 and not text_provided.isdigit():
+
+        all_additional_text = message['text'][6:]
+        if all_additional_text.strip() == 'help':
+            return self.reply(message, self.get_help(), opts)
+
+        how_many = all_additional_text[0:3].strip()
+        search_term = all_additional_text[3:].strip()
+
+        if len(how_many) > 0 and not how_many.isdigit():
             return self.reply(
                 message,
-                '\'{}\' is not a valid integer.'.format(text_provided),
+                '\'{}\' is not a valid integer.'.format(how_many),
                 opts
             )
 
-        how_many = int(text_provided or self.default_how_many)
+        how_many = (int(how_many)
+                    if len(how_many) > 0
+                    else self.default_how_many)
 
```

Comment:
Optional if you're up for it:

It'd be cool to split the message to start using something like `re.split(r'\s+', message['text'])` and then parsing the list that results. Seems like it might make for some cleaner code.

Reply:
Check out new utility I added
https://github.com/ReformedDevs/fraqbot/blob/master/fraqbot/Local/helpers/text.py#L123
https://github.com/ReformedDevs/fraqbot/blob/master/fraqbot/Local/newjob.py#L97

Reply:
Rock n roll